src/systemctl/systemctl-list-unit-files.c

```diff
@@ -103,7 +117,9 @@ static int output_unit_file_list(const UnitFileList *units, unsigned c) {
                 else
                         on_unit_color = on_underline;
 
-                id = basename(u->path);
+                r = path_extract_filename(u->path, &id);
+                if (r < 0)
+                        return table_log_add_error(r);
```

Comment:
that helper should only be used when you fail to add something to a table, i.e. if table_add() or its various flavours fail.

just do the usual `return log_error_errno(r, "…")` here.